src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/BuildHostProcessManager.cs

```diff
@@ -35,10 +35,17 @@ public async Task<IBuildHost> GetBuildHostAsync(string projectFilePath, Cancella
 
         _logger?.LogTrace($"Choosing a build host of type {neededBuildHostKind} for {projectFilePath}.");
 
+        if (neededBuildHostKind == BuildHostProcessKind.Mono && MonoMSBuildDiscovery.GetMonoMSBuildDirectory() == null)
```

Comment:
I think I'd prefer if we were consistent here and didn't look for mono in our host process.  And instead always delegated to `buildHost.HasUsableMSBuildAsync`

Reply:
If we're on Linux and we don't have a mono, then I can't even delegate to a host process in that case since I can't run one! And next week once I'm doing the work to move this down to MSBuildWorkspace itself we'll also get a similarish check for .NET Core as well, since you might be running on .NET Framework and we don't even know if there's a dotnet to launch to then call MSBuildLocator for.

Ideally MSBuildLocator would allow for non-matching runtimes and we would only have the host do the discovery; this launch-a-process-and-ask is really a hack around that at this point until we fix it.

Reply:
(and we don't have to have logic to discover a .NET Framework runtime, if only because it's a required Windows component...)

Reply:
> then I can't even delegate to a host process in that case since I can't run one!

ah good point.